controlpanel/api/auth0.py

```diff
@@ -0,0 +1,190 @@
+from collections import OrderedDict
+from urllib.parse import urljoin
+
+from auth0.v3 import authentication, exceptions
+from django.conf import settings
+import requests
+
+
+class Auth0Error(Exception):
+    pass
+
+
+class APIClient(object):
+    base_url = None
+    audience = None
+
+    def __init__(self, **kwargs):
+        self.client_id = kwargs.get("client_id", settings.AUTH0["client_id"])
+        self.client_secret = kwargs.get(
+            "client_secret", settings.AUTH0["client_secret"]
+        )
+        self.domain = kwargs.get("domain", settings.AUTH0["domain"])
+        self._access_token = None
```

Comment:
#### 1d. Refactor Auth0 client module

> Lines 1 - 190

The old client code was too indirect, and relied on subclassing `dict` and some difficult-to-follow inheritance and functional style.

Now we have a straightforward base class which provides methods to make authenticated API requests and wraps fetching all results across pagination...

[Next →](https://github.com/ministryofjustice/analytics-platform-control-panel/pull/610/files#diff-4abd63b561b5260e074659dad215fa1cR109)